core/src/avm2/globals/flash/net/socket.rs

```diff
@@ -0,0 +1,365 @@
+use crate::avm2::bytearray::Endian;
+use crate::avm2::error::{io_error, make_error_2008};
+pub use crate::avm2::object::socket_allocator;
+use crate::avm2::parameters::ParametersExt;
+use crate::avm2::{Activation, Error, Object, TObject, Value};
+use crate::context::UpdateContext;
+
+pub fn connect<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let socket = match this.as_socket() {
+        Some(socket) => socket,
+        None => return Ok(Value::Undefined),
+    };
+
+    let host = args.get_string(activation, 0)?;
+    let port = args.get_i32(activation, 1)? as u16;
+
```

Comment:
This should throw a `SecurityError`: `Error #2003: Invalid socket port number specified.`

Reply:
I used the TryFrom<u32> impl for u16 here. It errors when u32 is bigger than u16::MAX. Then I mapped the error to the correct SecurityError.